lib/client.js

```diff
@@ -139,7 +163,23 @@ module.exports = function (dependencies) {
           let error = new VError("stream ended unexpectedly");
           resolve({ device, error });
         }
-      })
+      });
+
+      request.setTimeout(this.config.requestTimeout, () => {
+        if (logger.enabled) {
+          logger('Request timeout');
+        }
+
+        resolve({ device, error: new VError("apn write timeout") });
+      });
+
+      request.on("aborted", () => {
```

Comment:
Probably useful to add for diagnostics - I assume end would get emitted after that

Reply:
Yes. That's the idea.